### ISSUE TITLE
"ValueError: Unallowed PCT %" when there's a "%" in the url

### ISSUE THREAD

**Issue by samuelcolvin** [aio-libs/yarl#124]
If there's a `%` (without it being a valid url encoded character) in the url path, yarl raises `ValueError: Unallowed PCT %` which causes a very unexpected 500 error.

Using `yarl==0.13.0`.

Traceback:

```
Error handling request
Traceback (most recent call last):
  File "/home/samuel/code/aiohttp-devtools/env/lib/python3.6/site-packages/aiohttp/web_protocol.py", line 275, in data_received
    messages, upgraded, tail = self._request_parser.feed_data(data)
  File "aiohttp/_http_parser.pyx", line 295, in aiohttp._http_parser.HttpParser.feed_data
  File "aiohttp/_http_parser.pyx", line 400, in aiohttp._http_parser.cb_on_header_field
  File "aiohttp/_http_parser.pyx", line 329, in aiohttp._http_parser.HttpRequestParserC._on_status_complete
  File "aiohttp/_http_parser.pyx", line 611, in aiohttp._http_parser._parse_url
  File "/home/samuel/code/aiohttp-devtools/env/lib/python3.6/site-packages/yarl/__init__.py", line 214, in build
    path = _quote(path, safe='@:', protected='/')
  File "yarl/_quoting.pyx", line 45, in yarl._quoting._quote
  File "yarl/_quoting.pyx", line 135, in yarl._quoting._do_quote
ValueError: Unallowed PCT %
```

Code to reproduce:

```python
from aiohttp import web

async def handle(request):
    return web.Response(text='<h1>testing</h1>', content_type='text/html')

app = web.Application()
app.router.add_get('/', handle)


if __name__ == '__main__':
    web.run_app(app, port=8001)
```

then curl

```
~ 0  151s ➤  curl -v "http://localhost:8001/%"
*   Trying 127.0.0.1...
* TCP_NODELAY set
* Connected to localhost (127.0.0.1) port 8001 (#0)
> GET /% HTTP/1.1
> Host: localhost:8001
> User-Agent: curl/7.52.1
> Accept: */*
> 
* HTTP 1.0, assume close after body
< HTTP/1.0 500 Internal Server Error
< Content-Type: text/html; charset=utf-8
< Content-Length: 141
< Date: Fri, 20 Oct 2017 16:11:26 GMT
< Server: Python/3.6 aiohttp/2.3.0
< 
* Curl_http_done: called premature == 0
* Closing connection 0
<html><head><title>500 Internal Server Error</title></head><body><h1>500 Internal Server Error</h1>Server got itself in trouble</body></html>~ 0  32ms ➤ 
```

**Comment by samuelcolvin** [comment]
I don't get this in production, I think because nginx is escaping the `%` before proxying the request.

**Comment by samuelcolvin** [comment]
Bump. @asvetlov: any update on this? I think it's quite important.

**Comment by asvetlov** [comment]
Ooohh! Thanks for pinging.
The issue is in my TODO list already but your request raised it to the top.

**Comment by asvetlov** [comment]
Hmm.
I have no idea how did you get the exception.
It is raised in `strict` mode only (off by default) but aiohttp itself never uses `strict` mode in own codebase.

**Comment by samuelcolvin** [comment]
Do you get the same error with my example above?

**Comment by asvetlov** [comment]
Yes.
I found the problem, will fix it.

**Comment by samuelcolvin** [comment]
👍

**Comment by asvetlov** [comment]
Fixed by #133
